iModelCore/Units/src/Formatting/NumericFormatSpec.cpp

```diff
@@ -959,4 +963,44 @@ double NumericFormatSpec::RoundDouble(double dval, double roundTo)
     return (dval < 0.0) ? -rnd : rnd;
     }
 
+Utf8String NumericFormatSpec::FormatToFractionalRatio(double dval) const
+{   
+    double reciprocal;
+    if (dval == 0.0)
+        reciprocal = 0.0;
+    else
+        reciprocal = 1.0 / dval;
+        
+    return "1:" + Format(reciprocal);
+}
+
+Utf8String NumericFormatSpec::FormatToIntegerRatio(double dval) const

```

Comment:
Part of this file was migrated from old C-code, so it uses some unnecessary c practices. We should not do that for new code. dVal is such a thing. Name the variable by what it represents, not by what it is (meaning value instead of dval)